### PR TITLE
Add a contact form

### PR DESCRIPTION
**Pull request by SFarida**
The major updates here are on the contact form and include : 
- validation for form fields, that is, required for name, email and description fields and a maxLength for the name and description fields.
- Collect data with Formspree service
- Update form CSS

### CODE REVIEW

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @SFarida, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 
-  Has a proper Pull Request. ✔️  
- No Linter errors :heavy_check_mark: 
- Good use of Github flow ✔️ 

### Changes Required

- [ ] I must commend you have done a great job working on the project and following the figma template chosen

- [ ] Kindly stick to the design in the Figma template given as current i can see some white space in your design on the 375px and 400px screen which are not in the figma template. Kindly fix it

Here is your design 👇 

far


- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.css [hunk 1]
```diff
@@ -1,6 +1,7 @@
 body {
   width: 375px;
   margin: 0;
+  padding: 0;
```
Comment (joskalenda) [inline index.css:4]: - [ ] Good job so far, but the changes you have added did not fix at it all, please consider making the requirement as mentioned in the previous

pv

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @SFarida , |
|Great work on making the changes requested by a previous reviewer 👏, You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. |
|  |

## Highlights

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.css [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+*,
+*::before,
+*::after {
+  margin: 0;
+  padding: 0;
+
+  /* no unit */
+  box-sizing: border-box;
+}
+
```
Comment (christianbiring1) [inline index.css]: - [ ] Kindly may consider removing the scroll on your page, The content is supposed to fits the screen size.
`Hint: You may consider giving your body a relative unit width of 100% and change all the sections you have used the absolute unit value of width (px) to %`

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @SFarida , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
# STATUS: REQUEST CHANGES :recycle: 
almost


 ## Highlights
- [ ] All linters are checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 

 ## Required Changes ♻️
- [ ]  `AS SUGGESTED BU THE TWO PREVIOUS REVIEWERS, KINDLY MAKE SURE YOUR PAGE WIDHT FITS WITH THE SCREEN SIZE WIDTH`
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SFarida** [comment]
Hello @christianbiring1 thank you for the review. I see that the white space issue was fixed as pointed out by the two previous reviewers before you. Also for the width I was ask to set it to 375px since the first week as it is on figma. 

Can you check this out please, I also think that the change about the scroll is not a required change and I did not add it in the file too. The code you commented on does not have scroll. I do not really understand why this is blocking my approval, please help me clarify this. Thank you

**Comment by christianbiring1** [comment]
> Hello @christianbiring1 thank you for the review. I see that the white space issue was fixed as pointed out by the two previous reviewers before you. Also for the width I was ask to set it to 375px since the first week as it is on figma.
> 
> Can you check this out please, I also think that the change about the scroll is not a required change and I did not add it in the file too. The code you commented on does not have scroll. I do not really understand why this is blocking my approval, please help me clarify this. Thank you

Hey @SFarida ,
Here's the screenshots of how your page looks since your last commit in this contact-form branch
Screenshot from 2022-12-07 10-21-25

Screenshot from 2022-12-07 11-15-36

**Comment by SFarida** [comment]
image

I do not understand why is is behaving that way because on my browser it is fitting, I will check it on other browsers again but please I do not think this is a change that should block the approval of this pull request.

**Comment by christianbiring1** [comment]
> image
> 
> I do not understand why it is behaving that way because on my browser it is fitting, I will check it on other browsers again but please I do not think this is a change that should block the approval of this pull request.

If you scroll on the right side you will see the `white space the previous reviewers were talking about, and that is what I have pointed out in my review as well.` on a screen size greater than `375px` there still be that white space.
As I have said you may consider using the relative unit value (`% or vw`) to set the width of your content on different sections where you have used the absolute unit value (`px`)

**Review by Lameck1 — COMMENTED** [review]
Hello @SFarida  👋,

_This is an additional review as a result of the second opinion request._

From my testing, it is true that there is that gap when your page is viewed on mobile screens larger than 375px. In responsive design, the page content should take the whole width of the viewport. As it is right now, your page will only look good on mobile screens with max width of 375px. This beats the goal of this exercise.

Please, for containers that you've given a fixed width of 375px, let their width be 100%. I have done this and everything looks good.

When done, please ask for another code review request from your dashboard.

Keep rocking 🚀!

File: index.css [hunk 3]
```diff
@@ -773,26 +736,25 @@ body {
...
 
   /* no unit */
   justify-content: center;
 }
 
 .footer_social_media {
-  align-self: center;
+  display: flex;
+  justify-content: center;
 }
```
Comment (OyePriscilla) [inline index.css:758]: - [optional] Kindly increase the size of the social media handles, I hardly see it. 👍 

## Your Design
 
socialmedia

**Review by OyePriscilla — APPROVED** [review]
Hi @SFarida   👍 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linters check passed ✔️ 
Good commit messages ✔️ 
GitHub flow observed 😄 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**